### PR TITLE
GNUMake: remove unused define for HIP compilation

### PR DESCRIPTION
**Pull request by mic84**
## Summary

* GNUMake: remove unused AMREX_HIP_PLATFORM define 

* GNUMake: rename HIP_PLATFORM to HIP_COMPILER for consistency 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

File: Tools/GNUMake/Make.defs [hunk 1]
```diff
@@ -42,12 +42,12 @@ ifeq ($(USE_HIP),TRUE)
...
-    $(error HIP_PLATFORM=hcc is no longer supported)
-  else ifeq ($(HIP_PLATFORM),clang)
+  HIP_COMPILER = $(shell hipconfig --compiler)
+  ifeq ($(HIP_COMPILER),nvcc)
+    $(error HIP_COMPILER=nvcc is not supported at this time. Use USE_CUDA to compile for NVIDIA platforms)
+  else ifeq ($(HIP_COMPILER),hcc)
+    $(error HIP_COMPILER=hcc is no longer supported)
+  else ifeq ($(HIP_COMPILER),clang)
   else
     $(error Unknown HIP_PLATFORM=$(HIP_PLATFORM))
```
Comment (WeiqunZhang) [inline Tools/GNUMake/Make.defs]: This needs to be updated too.

Comment (WeiqunZhang) [reply Tools/GNUMake/Make.defs]:
```Tools/GNUMake/comps/hip.mak:ifeq ($(HIP_PLATFORM),clang)
Tools/GNUMake/comps/hip.mak:else ifeq ($(HIP_PLATFORM),nvcc)
Tools/GNUMake/comps/hip.mak:  $(error HIP_PLATFORM nvcc is not supported at this time. Use USE_CUDA to compile for NVIDIA platforms.)
```

Comment (mic84) [reply Tools/GNUMake/Make.defs]: Now it should be good to go

Comment (WeiqunZhang) [reply Tools/GNUMake/Make.defs]: Seems that there still is one place in Make.defs.

**Review by ax3l — APPROVED** [review]
Thanks! :)

Edit: one last location update to go :)

File: Tools/GNUMake/Make.defs (hunk 1 above)
Comment (WeiqunZhang) [reply Tools/GNUMake/Make.defs]: `$(error Unknown HIP_PLATFORM=$(HIP_PLATFORM))` is still there, I think

Comment (ax3l) [reply Tools/GNUMake/Make.defs]: Can confirm, last one:
```
Tools/GNUMake/Make.defs
52:    $(error Unknown HIP_PLATFORM=$(HIP_PLATFORM))
```

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**